### PR TITLE
process-manager unit tests

### PR DESCRIPTION
**Pull request by NehaNaithani**

### CODE REVIEW

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/DockerUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,1135 @@
...
+
+    /**
+     * Test getDockerBridgeName when network.getOptions() is empty
+     */
+    @Test
+    public void tesGetDockerBridgeNameWhenNewtorkIsEmpty() {
+        assertNull(dockerUtil.getDockerBridgeName());
+    }
+
+    /**
```
Comment (baghbidi) [inline iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/DockerUtilTest.java]: why is it commented out?

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,552 @@
...
+     * TasK is remove And microservice is Empty
+     */
+    @Test
+    public void testExecuteWhenContainerTaskIsNotNullAndMicroserviceIsEmpty() {
+        try {
+            PowerMockito.when(containerTask.getAction()).thenReturn(ContainerTask.Tasks.REMOVE);
+            containerManager.execute(containerTask);
+            PowerMockito.verifyStatic(DockerUtil.class);
+            DockerUtil.getInstance();
+            PowerMockito.verifyPrivate(containerManager).invoke("removeContainerByMicroserviceUuid", any(), anyBoolean());
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,552 @@
...
+    /**
+     * Test execute when containerTask is null
+     */
+    @Test
+    public void testExecuteWhenContainerTaskIsNull() {
+        try {
+            containerManager.execute(null);
+            PowerMockito.verifyStatic(DockerUtil.class);
+            DockerUtil.getInstance();
+            PowerMockito.verifyStatic(LoggingService.class);
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java [hunk 4]
```diff
@@ -0,0 +1,552 @@
...
+        }
+    }
+
+    /**
+     * Test execute when containerTask is not null
+     * TasK is remove And microservice is not empty
+     * Task contains microserviceId which is not valid or already removed
+     */
+    @Test
+    public void testExecuteWhenContainerTaskRemoveMicroserviceIdNotValidAndMicroserviceIsNotEmpty() {
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java [hunk 5]
```diff
@@ -0,0 +1,552 @@
...
+        }
+    }
+
+    /**
+     * Test execute when containerTask is not null
+     * TasK is remove And microservice is not empty
+     * Task contains microserviceId which is valid and not already removed
+     */
+    @Test
+    public void testExecuteWhenContainerTaskRemoveMicroserviceIdIsValid() {
```
[the remark this answers is not in the document]
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/ContainerManagerTest.java]: updated

File: iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/DockerUtilTest.java (hunk 1 above)
Comment (NehaNaithani) [reply iofog-agent-daemon/src/test/java/org/eclipse/iofog/process_manager/DockerUtilTest.java]: Removed not needed

**Pull request merged**